src/transformers/trainer_callback.py

```diff
@@ -64,7 +64,10 @@ class TrainerState:
             to avoid overflow).
         log_history (`List[Dict[str, float]]`, *optional*):
             The list of logs done since the beginning of training.
-        best_metric (`float`, *optional*):
+        best_metrics (`Dict[str, float]`, *optional*):
+            When tracking the best model, the full metrics dictionary for the checkpoint with the best metric value
+            encounted so far.
+        best_metric_value (`float`, *optional*):
```

Comment:
These changes the previous type of `best_metrics` which is not good. I understand you need to return the full metric dictionary for use in `run_hp_search_xxx`. May keep the previous `best_metric` and have a new entry named `best_metrics_dict`

Reply:
Gotcha. I'll keep the old `best_metric`. I thought you might want something a more specific name here (since `metric_value` is used everywhere outside of the state class). But I realize now that this might be a breaking change.